pkg/ingester/ingester_v2.go

```diff
@@ -415,6 +416,9 @@ type TSDBState struct {
 	appenderAddDuration    prometheus.Histogram
 	appenderCommitDuration prometheus.Histogram
 	idleTsdbChecks         *prometheus.CounterVec
+
+	closed bool // protected by userStatesMtx
```

Comment:
I am wondering if we should use an extra boolean flag, or rely on Stopping state of ingester. Extra boolean flag allows us to reply to query functions while compaction and shipping loops are still running, but in practice I'm not sure if it makes enough difference compared to just using ingester's state.

Therefore I would suggest we remove this flag. We can already check for ingester's state (`i.State() == services.Stopping`) and reject queries in that case. Ingester only calls `closeAllTSDB` after it enters `Stopping` state (and after its subservices have finished).

Reply:
I agree, I don't think it is necessary to have the extra boolean. I've removed the `closed bool` and added the check as per your suggestion.